trace-utils/src/msgpack_decoder/decode/string.rs

```diff
@@ -89,10 +130,21 @@ where
 {
     let slice = unsafe { buf.as_mut_slice() };
 
-    if slice.first() == Some(NULL_MARKER) {
-        *slice = &slice[1..];
+    if is_null_marker(slice) {
         Some(default())
     } else {
         None
     }
 }
+
+/// When you want to "peek" if the next value is a null marker, and only advance the buffer if it is
+/// null. If it is not null, you can continue to decode as expected.
+#[inline]
+pub fn is_null_marker(buf: &mut &[u8]) -> bool {
```

Comment:
nit: the name doesn't really show that this can advance the buffer

```suggestion
pub fn consume_nil_marker(buf: &mut &[u8]) -> bool {
```

Reply:
I reverted to `handle_null_marker` since I removed the old one.